include/gbt.h

```diff
@@ -11,96 +11,67 @@
  *  Each word is the number of time the scan was "good" out of the total number of scan requested, N.
  *
  */
-bool scanGBTPhasesLocal(localArgs *la, const uint32_t ohN, const uint32_t N = 1, const uint8_t phaseMin = gbt::PHASE_MIN, const uint8_t phaseMax = gbt::PHASE_MAX, const uint8_t phaseStep = 1);
+struct scanGBTPhases : public xhal::rpc::Method
 
-/*! \brief Write the GBT configuration of one OptoHybrid.
- *  \param[in] request RPC response message.
- *  \param[out] response RPC response message.
- *
- *  The method expects the following RPC keys :
- *  - `word ohN` : OptoHybrid index number.
- *  - `word gbtN` : Index of the GBT to configure. There 3 GBT's per OptoHybrid in the GE1/1 chambers.
- *  - `binarydata config` : Configuration blob of the GBT. This is a 366 element long vector of the registers to write. The registers are sorted from address 0 to address 365.
- *
- *  The method returns the following RPC keys :
- *  - `string error` : If an error occurs, this keys exists and contains the error message.
- */
-void writeGBTConfig(const RPCMsg *request, RPCMsg *response);
+{
+
+    std::vector<std::vector<uint32_t>> operator()(const uint32_t &ohN, const uint32_t &N = 1, const uint8_t &phaseMin = gbt::PHASE_MIN, const uint8_t &phaseMax = gbt::PHASE_MAX, const uint8_t &phaseStep = 1) const;
+
+};
 
-/*! \brief Local callable version of `writeGBTConfig`
- *  \param[in, out] la Local arguments structure.
+
+/*! \brief Scan the GBT phases of one OptoHybrid.
  *  \param[in] ohN OptoHybrid index number.
  *  \param[in] gbtN Index of the GBT to write. There 3 GBT's per OptoHybrid in the GE1/1 chambers.
  *  \param[in] config Configuration blob of the GBT. This is a 366 elements long array whose each element is the value of one register sorted from address 0 to address 365.
- *  \return Returns `false` in case of success; `true` in case of error. The precise error is logged and written to the `error` RPC key.
+ *  \return Returns `false` in case of success; `true` in case of error. The precise error is logged and written to the `LOG4CPLUS` logger.
  */
-bool writeGBTConfigLocal(localArgs *la, const uint32_t ohN, const uint32_t gbtN, const gbt::config_t &config);
+struct writeGBTConfig : public xhal::rpc::Method
 
-/*! \brief Write the phase of a single VFAT.
- *  \param[in] request RPC response message.
- *  \param[out] response RPC response message.
- *
- *  The method expects the following RPC keys :
- *  - `word ohN` : OptoHybrid index number.
- *  - `word vfatN` : Index of the VFAT to configure. There 24 VFAT's per OptoHybrid in the GE1/1 chambers.
- *  - `word phase` : Phase value to write. The values span from 0 to 15.
- *
- *  The method returns the following RPC keys :
- *  - `string error` : If an error occurs, this keys exists and contains the error message.
- */
-void writeGBTPhase(const RPCMsg *request, RPCMsg *response);
+{
+
+    bool operator()(const uint32_t &ohN, const uint32_t &gbtN) const;
```

Comment:
here i think you're missing the data to be written to the registers (`gbt::config_t` in the old implementation)

Reply:
From what I could tell in the old implementation, the ``gbt::config_t`` was an input for the local function but was called within the nonlocal RPC implementation, so instead I just moved it internal to ``writeGBTConfig`` since the configs it accesses seem to be constants.

Reply:
the docstring on the old line 57 shows that this configuration data was passed in to the RPC method (the `binarydata` `config` blob), now, however, this data will be directly passed to the functor.
see https://github.com/jsturdy/ctp7_modules/pull/2/files/bfb5d36c03787003292811cee87cf8b1e5c31d96#r345152651
and https://github.com/jsturdy/ctp7_modules/pull/2/files/bfb5d36c03787003292811cee87cf8b1e5c31d96#r345152760